### PR TITLE
Create sieveOfEratosthenes.java issue number #308

### LINKED ISSUE #308: Java implementation of Sieve of Eratosthenes to generate prime numbers.
Please include a small description or summary of the issue here.

**Is your feature request related to a problem? Please describe.**
It will help people to understand the implementation Sieve of Eratosthenes to generate prime numbers in O(n*log(log(n))) time complexity.

**Describe the solution you'd like**
A standard implementation of Sieve of Eratosthenes.

**Languages**

-   [ ] Python
-   [x] Java
-   [ ] C
-   [ ] C++

### PR DESCRIPTION
**Pull request by prateekrai123**
Sieve of Eratosthenes implementation in java
Please include a summary of the changes you have made in regard with this issue. 

Fixes #308

## Type of change

**Please delete options that are not relevant**

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] This change requires a documentation update

# Checklist:

-   [x] I have read the contribution guidelines before making this PR.
-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented in my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation (Language folder or any other readme instructed to)
-   [x] My changes generates no new warnings
-   [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by prateekrai123** [comment]
@kaustubhgupta please review. In reference to issue #308

File: Java/sieveOfEratosthenes.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import java.util.Scanner;
+
+// Author -> Prateek Kumar Rai
```
Comment (kaustubhgupta) [inline Java/sieveOfEratosthenes.java]: Remove these commets

File: Java/sieveOfEratosthenes.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+// Github -> github.com/prateekrai123
+
+public class sieveOfEratosthenes {
+
+	public static void main(String[] args) {
+
```
Comment (kaustubhgupta) [inline Java/sieveOfEratosthenes.java:6]: Remove extra spaces from the program

File: Java/sieveOfEratosthenes.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+		}
+		
+		//Printing all prime numbers till num
+		
+		for(int i=2;i<=num;i++) {
+			
+			if(prime[i]) {
+				
+				System.out.print(i+" ");
+
```
Comment (kaustubhgupta) [inline Java/sieveOfEratosthenes.java]: Update Java folder readme

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

File: Java/sieveOfEratosthenes.java (hunk 2 above)
Comment (prateekrai123) [reply Java/sieveOfEratosthenes.java:6]: Sure.

File: Java/sieveOfEratosthenes.java (hunk 1 above)
Comment (prateekrai123) [reply Java/sieveOfEratosthenes.java]: Sure.

File: Java/sieveOfEratosthenes.java (hunk 3 above)
Comment (prateekrai123) [reply Java/sieveOfEratosthenes.java]: Okay.

**Comment by prateekrai123** [comment]
@kaustubhgupta done. Please check it.

**Pull request closed without merging**